plugins/gradle/src/main/java/org/wildfly/swarm/plugin/gradle/PackageTask.java

```diff
@@ -66,6 +71,23 @@ public void packageForSwarm() throws Exception {
 
         GradleArtifactResolvingHelper resolvingHelper = new GradleArtifactResolvingHelper(project);
         Properties propertiesFromExtension = getPropertiesFromExtension();
+        List<File> moduleDirs = getModuleDirs();
+        if (moduleDirs.isEmpty()) {
```

Comment:
You can check the original code.
Before
Case modulesDir is not set.
When SwarmExtension is created, it will added the default from resourceOutputDir.
It will use the default. But the situation only works war is run before.
Case modulesDir is set
Clear the list which is initialized when SwarmExtension is created, add the new value to the list.
If will use the new value from it.

After
Case modulesDir is not set.
modulesDir is empty. it will added the default from resourceDir when the task run. The only change is that it's in the task instead of creation of SwarmExtension.
Case modulesDIr is set
modulesDIr is set to new value. The same as it previously does. The only difference is when the new value is empty list. It will use default. But it's meaningless setting an empty list, right?